applications/FluidDynamicsApplication/custom_elements/alternative_qs_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,685 @@
+    const auto& fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+    BoundedMatrix<double,Dim,Dim> permeability = this->GetAtCoordinate(rData.Permeability, rData.N);
+    BoundedMatrix<double,Dim,Dim> sigma = ZeroMatrix(Dim, Dim);
+    const auto& r_body_forces = rData.BodyForce;
+    const auto& r_velocities = rData.Velocity;
+    const auto& r_pressures = rData.Pressure;
+    const auto& fluid_fraction_gradient = this->GetAtCoordinate(rData.FluidFractionGradient, rData.N);
+
+    double det_permeability = MathUtils<double>::Det(permeability);
+    MathUtils<double>::InvertMatrix(permeability, sigma, det_permeability, -1.0);
+    sigma *= viscosity;
+
+    for (unsigned int i = 0; i < NumNodes; i++) {
+            const array_1d<double,3>& r_acceleration = rGeom[i].FastGetSolutionStepValue(ACCELERATION);
+            Vector sigma_U = ZeroVector(Dim);
+            BoundedMatrix<double,Dim,Dim> sym_gradient_u = ZeroMatrix(Dim, Dim);
+            Vector grad_alpha_sym_grad_u = ZeroVector(Dim);
+            for (unsigned int d = 0; d < Dim; d++) {
+                double div_u = 0.0;
+                for (unsigned int e = 0; e < Dim; e++){
+                    sigma_U[d] += sigma(d,e) * rData.N[i] * r_velocities(i,e);
+                    sym_gradient_u(d,e) += 1.0/2.0 * (rData.DN_DX(i,d) * r_velocities(i,e) + rData.DN_DX(i,e) * r_velocities(i,d));
+                    grad_alpha_sym_grad_u[d] += fluid_fraction_gradient[d] * sym_gradient_u(d,e);
+                    div_u += rData.DN_DX(i,e) * r_velocities(i,e);
+                }
+                rResidual[d] += density * (rData.N[i] * r_body_forces(i,d) - fluid_fraction * rData.N[i] * r_acceleration[d] - fluid_fraction * convection[i] * r_velocities(i,d)) + 2 * grad_alpha_sym_grad_u[d] * viscosity - 2.0 / 3.0 * viscosity * fluid_fraction_gradient[d] * div_u - fluid_fraction * rData.DN_DX(i,d) * r_pressures[i] - sigma_U[d];
+            }
+        }
+}
+
+template< class TElementData >
+void AlternativeQSVMSDEMCoupled<TElementData>::MomentumProjTerm(
+    const TElementData& rData,
+    const array_1d<double,3>& rConvectionVelocity,
+    array_1d<double,3> &rMomentumRHS) const
+{
+    Vector AGradN;
+    this->ConvectionOperator(AGradN,rConvectionVelocity,rData.DN_DX);
+
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    const double viscosity = this->GetAtCoordinate(rData.DynamicViscosity, rData.N);
+    const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+    BoundedMatrix<double,Dim,Dim> permeability = this->GetAtCoordinate(rData.Permeability, rData.N);
+    BoundedMatrix<double,Dim,Dim> sigma = ZeroMatrix(Dim, Dim);
+
+    const auto& r_body_forces = rData.BodyForce;
+    const auto& r_velocities = rData.Velocity;
+    const auto& r_pressures = rData.Pressure;
+    const auto& fluid_fraction_gradient = this->GetAtCoordinate(rData.FluidFractionGradient, rData.N);
```

Comment:
```suggestion
    const auto& r_fluid_fraction_gradient = this->GetAtCoordinate(rData.FluidFractionGradient, rData.N);
```
(minor style guide suggestion)